pkgs/top-level/darwin-packages.nix

```
@@ -240,5 +241,8 @@ impure-cmds // appleSourcePackages // chooseLibs // {
       };
 
     in
-      nixos.config.system.build.macos-builder-installer;
+      nixos.config.system.build.macos-builder-installer) { modules = [ ]; };
+
+} // lib.optionalAttrs config.allowAliases {
+  builder = throw "'darwin.builder' has been renamed to 'darwin.linux-builder' and requires a migration: https://nixos.org/manual/nixpkgs/unstable/#sec-darwin-builder"; # added 2023-07-04
```

They wouldn't know what to look for, but we can make them aware of the change here and still refer to the instructions for details.

```suggestion
  builder = throw "'darwin.builder' has been changed and renamed to 'darwin.linux-builder'. The default ssh port is now 31022. Please update your configuration or override the port back to 22. See https://nixos.org/manual/nixpkgs/unstable/#sec-darwin-builder"; # added 2023-07-04
```

---

Updated :+1: